gabbi/json_parser.py

```diff
@@ -0,0 +1,51 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import jsonpath_rw
+
+
+class Len(jsonpath_rw.JSONPath):
+    """The JSONPath referring to the len of the current object.
+
+    Concrete syntax is '`len`'.
+    """
+
+    def find(self, datum):
+        datum = jsonpath_rw.DatumInContext.wrap(datum)
+        return [jsonpath_rw.DatumInContext(len(datum.value),
```

Comment:
I'm not sure if it should, but does it make any sense for it also to work on str and dict?

In fact, would it make any sense to just call len and handle TypeError if the datum.value can't len()?

I can certainly imagine situations where people will want to know the length of a dict that is being returned.

What do you think?

Bringing in @FND as he may have an opinion too. FND, in case you feel like this is out of the blue, especially with regard to discussions we've had about dealing with location, there's some precedent, I've been looking for months to [add sorts](https://github.com/kennknowles/python-jsonpath-rw/pull/25) but hadn't figured out the best way to integrate. This gives us a place to hang such things. _If_ we choose to do so.